### PR TITLE
Configure CI

### PR DESCRIPTION
**Pull request by xirc**
Requirements:
* [x] Test against JDK8 and JDK11
* [x] Cache
* [x] Enable scalac option "-Xfatal-warnings" only on CI
* [x] Publish test report
* [x] Check minimum coverage

We will use the following GitHub actions:
* https://github.com/actions/checkout
* https://github.com/actions/setup-java
* https://github.com/coursier/cache-action
* https://github.com/mikepenz/action-junit-report

Notes:
* We have to detect binary backward compatibility break after stable version release.

Tests:
* [[DONT MERGE] CI fails if a unit test fails by xirc · Pull Request #2 · lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy](https://github.com/lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy/pull/2)
* [[DONT MERGE] CI fails if a compiler warning occurs by xirc · Pull Request #3 · lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy](https://github.com/lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy/pull/3)
* [[DONT MERGE] CI fails if coverage doesn't meet the minimum criteria by xirc · Pull Request #4 · lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy](https://github.com/lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy/pull/4)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,11 +6,19 @@ ThisBuild / scalacOptions ++= Seq(
...
+ThisBuild / scalacOptions ++= sys.props
+  .get("lerna.enable.discipline")
+  .map(_ => "-Xfatal-warnings")
+  .toSeq
 ThisBuild / autoAPIMappings := true
 
+// Minimum Coverage
+ThisBuild / coverageFailOnMinimum := true
+ThisBuild / coverageMinimumStmtTotal := 80
+ThisBuild / coverageMinimumBranchTotal := 80
```
Comment (xirc) [inline build.sbt:20]: NOTE: These settings can be found [here](https://github.com/scoverage/sbt-scoverage#minimum-coverage).

**Comment by negokaz** [comment]
@xirc 

How about adding a badge to the README.md to notify CI status?

**example) A badge to notify CI status of main branch:**

[CI](https://github.com/lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy/actions/workflows/ci.yml?query=workflow%3ACI+branch%3Amain)
```
[![CI](https://github.com/lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy/actions/workflows/ci.yml/badge.svg?branch=main)](https://github.com/lerna-stack/akka-coordinator-avoidance-shard-allocation-strategy/actions/workflows/ci.yml?query=workflow%3ACI+branch%3Amain)
```

**Comment by xirc** [comment]
@negokaz Thanks! That's great. I've added a badge to README.

**Review by negokaz — APPROVED** [review]
LGTM 👍

**Pull request merged**